core/storage/btree.rs

```diff
@@ -1555,23 +1552,16 @@ fn find_free_cell(page_ref: &PageContent, db_header: Ref<DatabaseHeader>, amount
 
 impl Cursor for BTreeCursor {
     fn seek_to_last(&mut self) -> Result<CursorResult<()>> {
-        self.move_to_rightmost()?;
-        match self.get_next_record(None)? {
-            CursorResult::Ok((rowid, next)) => {
-                if rowid.is_none() {
-                    match self.is_empty_table()? {
-                        CursorResult::Ok(is_empty) => {
-                            assert!(is_empty)
-                        }
-                        CursorResult::IO => (),
-                    }
-                }
-                self.rowid.replace(rowid);
-                self.record.replace(next);
-                Ok(CursorResult::Ok(()))
-            }
-            CursorResult::IO => Ok(CursorResult::IO),
+        return_if_io!(self.move_to_rightmost());
```

Comment:
@pereman2 before i merge this can you confirm that this was a bug that it didnt check the result for io?

Reply:
looks like it XD